R/output.R

```diff
@@ -86,16 +84,18 @@ Output <- R6::R6Class(
       self$dataDisplayName <- dataDisplayName %||% self$displayName
       self$residualScale <- residualScale
 
-      # If filter is null, assumes that user won't get any observed data
+      # If data selection is expression, use it as is
       self$dataSelection <- dataSelection
       # Ensure that dataFilter is of type expression
       if (isOfType(self$dataSelection, "character")) {
-        if (self$dataSelection %in% DataSelectionKeys$ALL) {
-          self$dataSelection <- "TRUE"
-        }
-        self$dataSelection <- parse(text = self$dataSelection)
-        if (dataSelection %in% DataSelectionKeys$NONE) {
-          self$dataSelection <- NULL
+        # When concatenating, ALL won't be understood by dplyr
+        # Needs to be replaced by true to select all data
+        dataSelection[dataSelection %in% DataSelectionKeys$ALL] <- TRUE
+        # Concatenate selections using &
+        self$dataSelection <- paste(dataSelection, collapse = " & ")
+        # If any selection include None, do not select anything
+        if (isIncluded(DataSelectionKeys$NONE, dataSelection)) {
+          self$dataSelection <- FALSE
```

Comment:
brackets are missing when concatenating with &.

E.g. consider the case `dataSelection=c('2<3 | 4<3 ', '5<3')` 
The expectation is FALSE, because the last term `'5<3'` is FALSE. But:
```
> paste(dataSelection, collapse = " & ")
[1] "2<3 | 4<3  & 5<3"
> 2<3 | 4<3  & 5<3
[1] TRUE
```

So if the brackets are not somehow added before the function call: they must be added here